src/daft-dsl/src/expr/mod.rs

```diff
@@ -216,8 +216,13 @@ pub enum Expr {
         inputs: Vec<ExprRef>,
     },
 
+    // Over represents a window function as it is actually evaluated (since it requires a window spec)
+    #[display("{_0} over {_1}")]
+    Over(WindowExpr, window::WindowSpec),
+
+    // WindowFunction represents a window function as an expression, this alone cannot be evaluated since it requires a window spec
```

Comment:
Let's add a comment here to explain that this is really only used for constructing window functions in dataframe, and this enum variant should otherwise not exist